lib/resty/kafka/producer.lua

```diff
@@ -62,7 +62,7 @@ end
 
 local function produce_encode(self, topic_partitions)
     local req = request:new(request.ProduceRequest,
-                            correlation_id(self), self.client.client_id, self.api_version)
+                            correlation_id(self), self.client.client_id, request.API_VERSION_V1)
```

Comment:
nope, we should not hard code the API version.
please fix the other hard code places.

Reply:
And we should check the api_version should >=1 when it specified the sasl_config during new a producer.